### DISCUSSION TITLE
Unneeded(?) dependencies on System.Text.Json package

### DISCUSSION THREAD

**Discussion by mus65** [dotnet/runtime#105014] [General]
There have already been some issues regarding transitive dependencies to the the vulnerable System.Text.Json 8.0.0 (e.g.#104619,  #104705, #104669).

My question is: since System.Text.Json is shipped inbox with .NET itself, why do e.g. net8.0 targeted assemblies even depend on the System.Text.Json package? Is there a technical reason for this?

Removing the dependency altogether would avoid a lot of false positives from NuGet audit and avoid the chore of keeping the package up-to-date.

Some affected packages that we noticed in our project (I assume there are a lot more):

[Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console#dependencies-body-tab)
[Microsoft.Extensions.Configuration.Json](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Json#dependencies-body-tab)
[System.Memory.Data](https://www.nuget.org/packages/System.Memory.Data#dependencies-body-tab)

**Comment by huoyaoyuan** [comment]
See #95626 and #97937

**Comment by mus65** [comment]
This was actually fixed by #106172
